From 5f37cf7f021c0f939a4f04a333cccdc6f25783e2 Mon Sep 17 00:00:00 2001
From: Patrick Spendrin <ps_ml@gmx.de>
Date: Mon, 28 Apr 2014 10:35:06 +0200
Subject: [PATCH 14/28] prevent ex/import in exported classes used in tests

If using class definitions of exported classes in tests,
the export/import mechanism isn't needed. To prevent
linker errors, set the EXPORT macro to empty.
---
 src/file/autotest/CMakeLists.txt                 | 4 ++--
 src/file/autotest/fileindexingjob/CMakeLists.txt | 2 +-
 src/file/lib/autotests/CMakeLists.txt            | 4 ++--
 src/file/search/autotest/CMakeLists.txt          | 2 ++
 src/file/tests/CMakeLists.txt                    | 4 ++--
 5 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/src/file/autotest/CMakeLists.txt b/src/file/autotest/CMakeLists.txt
index a54736e..d909c65 100644
--- a/src/file/autotest/CMakeLists.txt
+++ b/src/file/autotest/CMakeLists.txt
@@ -26,7 +26,7 @@ set(metadatamover_SRC
     ../lib/filemapping.cpp)
 
 kde4_add_unit_test(metadatamovertest NOGUI ${metadatamover_SRC})
-
+set_target_properties(metadatamovertest PROPERTIES COMPILE_FLAGS -DBALOO_FILE_EXPORT=)
 target_link_libraries(metadatamovertest
   ${QT_QTTEST_LIBRARY}
   ${QT_QTSQL_LIBRARY}
@@ -83,7 +83,7 @@ set(basicIndexingQueue_SRC
 )
 
 kde4_add_unit_test(basicindexingqueuetest NOGUI ${basicIndexingQueue_SRC})
-
+set_target_properties(basicindexingqueuetest PROPERTIES COMPILE_FLAGS -DBALOO_FILE_EXPORT=)
 target_link_libraries(basicindexingqueuetest
   ${QT_QTTEST_LIBRARY}
   ${QT_QTSQL_LIBRARY}
diff --git a/src/file/autotest/fileindexingjob/CMakeLists.txt b/src/file/autotest/fileindexingjob/CMakeLists.txt
index 811ed30..3fc5b2b 100644
--- a/src/file/autotest/fileindexingjob/CMakeLists.txt
+++ b/src/file/autotest/fileindexingjob/CMakeLists.txt
@@ -10,7 +10,7 @@ target_link_libraries(baloo_file_extractor_dummy
 add_definitions("-DBALOO_TEMP_PATH=\\\"${CMAKE_CURRENT_BINARY_DIR}\\\"")
 kde4_add_unit_test(fileindexingjobtest NOGUI fileindexingjobtest.cpp
                                              ../../fileindexingjob.cpp)
-
+set_target_properties(fileindexingjobtest PROPERTIES COMPILE_FLAGS -DBALOO_FILE_EXPORT=)
 target_link_libraries(fileindexingjobtest
   ${QT_QTTEST_LIBRARY}
   ${KDE4_KDECORE_LIBS}
diff --git a/src/file/lib/autotests/CMakeLists.txt b/src/file/lib/autotests/CMakeLists.txt
index 95a0916..cfcb390 100644
--- a/src/file/lib/autotests/CMakeLists.txt
+++ b/src/file/lib/autotests/CMakeLists.txt
@@ -47,8 +47,8 @@ target_link_libraries(modifyjobtest
 
 # Pass CMAKE_CURRENT_BINARY_DIR so tests can use it to create some of the
 # temporary files they need (as XattrDetector blacklists /tmp by default).
-set_target_properties(fetchjobtest PROPERTIES COMPILE_FLAGS -DBUILDDIR="\\"${CMAKE_CURRENT_BINARY_DIR}/\\"")
-set_target_properties(modifyjobtest PROPERTIES COMPILE_FLAGS -DBUILDDIR="\\"${CMAKE_CURRENT_BINARY_DIR}/\\"")
+set_target_properties(fetchjobtest PROPERTIES COMPILE_FLAGS -DBUILDDIR="\\"${CMAKE_CURRENT_BINARY_DIR}/\\"" -DBALOO_FILE_EXPORT=)
+set_target_properties(modifyjobtest PROPERTIES COMPILE_FLAGS -DBUILDDIR="\\"${CMAKE_CURRENT_BINARY_DIR}/\\"" -DBALOO_FILE_EXPORT=)
 
 #
 # TagList Job
diff --git a/src/file/search/autotest/CMakeLists.txt b/src/file/search/autotest/CMakeLists.txt
index eb058d2..836ef2b 100644
--- a/src/file/search/autotest/CMakeLists.txt
+++ b/src/file/search/autotest/CMakeLists.txt
@@ -9,6 +9,8 @@ kde4_add_unit_test(filesearchstoretest filesearchstoretest.cpp
                    ../../database.cpp
 )
 
+set_target_properties(filesearchstoretest PROPERTIES COMPILE_FLAGS -DBALOO_FILE_EXPORT=)
+
 target_link_libraries(filesearchstoretest
   ${QT_QTCORE_LIBRARY}
   ${QT_QTSQL_LIBRARY}
diff --git a/src/file/tests/CMakeLists.txt b/src/file/tests/CMakeLists.txt
index 102c981..7b914b2 100644
--- a/src/file/tests/CMakeLists.txt
+++ b/src/file/tests/CMakeLists.txt
@@ -27,7 +27,7 @@ kde4_add_executable(basicindexingqueuetest
                     ../lib/filemapping.cpp
                     ../commitqueue.cpp
 )
-
+set_target_properties(basicindexingqueuetest PROPERTIES COMPILE_FLAGS -DBALOO_FILE_EXPORT=)
 target_link_libraries(basicindexingqueuetest
   ${QT_QTCORE_LIBRARY}
   ${QT_QTSQL_LIBRARY}
@@ -52,7 +52,7 @@ kde4_add_executable(fileindexingqueuetest
                     ../lib/filemapping.cpp
                     ../commitqueue.cpp
 )
-
+set_target_properties(fileindexingqueuetest PROPERTIES COMPILE_FLAGS -DBALOO_FILE_EXPORT=)
 target_link_libraries(fileindexingqueuetest
   ${QT_QTCORE_LIBRARY}
   ${QT_QTSQL_LIBRARY}
-- 
1.9.0

